app/controllers/planning_applications/charges_controller.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+module PlanningApplications
+  class ChargesController < AuthenticationController
+    before_action :set_planning_application
+
+    def index
+      respond_to do |format|
+        format.html
+      end
+    end
+
+    def new
+      @charge = @planning_application.charges.build
+      @payment = @charge.build_payment
+
+      respond_to do |format|
+        format.html
+      end
+    end
+
+    def create
+      @charge = @planning_application.charges.new(charge_params)
+      if @charge.save
+        redirect_to planning_application_charges_path(@planning_application), notice: "Charge created successfully."
+      else
+        @charge.build_payment unless @charge.payment
+        render :new
+      end
+    end
+
+    def edit
+      @charge = @planning_application.charges.find(params[:id])
```

Comment:
Perhaps this line could be extracted into a`set_charge` method, then `before_action :set_change, only: %[edit update destroy]`